src/topmodel.c

```diff
@@ -508,9 +508,189 @@ if(yes_print_output==TRUE)
+extern void convert_dist_to_histords(double *dist_from_outlet, int num_channels,
+					double chv, double rv, double dt, double tch[11])
+{
+    // declare local variables
+    double chvdt, rvdt;
+    int j;
+    
+    /*  Convert parameters to m/time step DT
+    *  with exception of XK0 which must stay in m/h
+    *                    Q0 is already in m/time step
+                     T0 is input as Ln(To) */
+
+    chvdt = chv*dt;
+    rvdt = rv*dt;
+
+    tch[1]=dist_from_outlet[1]/chvdt;
+    for(j=2;j<=num_channels;j++)
+     {
+      tch[j]=tch[1]+(dist_from_outlet[j]-dist_from_outlet[1])/rvdt;
+     }
+
+    return;
+
+}
+
+// FUNCTION TO CALCULATE THE TIME_DELAY_HISTOGRAM
+// updates values associated with model structure
+// outputs used in TOPMOD() include num_time_delay_histo_ords, time_delay_histogram,
+// num_delay, and Q
+extern void calc_time_delay_histogram(int max_time_delay_ordinates, int num_channels, double area,
+				double tch[11], int *num_time_delay_histo_ords, int *num_delay,
+				double **time_delay_histogram, double *cum_dist_area_with_dist) 
+
+
+{
+
+    // declare local variables
+    double time, a1, sumar, a2;
+    int j, ir;
+
+    if((*time_delay_histogram)==NULL)
+      {
+      d_alloc(time_delay_histogram,max_time_delay_ordinates);
+      }
+
+
+    (*num_time_delay_histo_ords)=(int)tch[num_channels];
+    
+    
+    if((double)(*num_time_delay_histo_ords)<tch[num_channels]) 
+       {
+       (*num_time_delay_histo_ords)++;
+       }
+    
+    (*num_delay)=(int)tch[1];
+    (*num_time_delay_histo_ords)-=(*num_delay);
+    
+
+
+    for(ir=1;ir<=(*num_time_delay_histo_ords);ir++)
+      {
+      time=(double)(*num_delay)+(double)ir;
+      if(time>tch[num_channels])
+        {
+        (*time_delay_histogram)[ir]=1.0;
+        }
+      else
+        {
+        for(j=2;j<=num_channels;j++)
+          {
+          if(time<=tch[j])
+    	{
+    	(*time_delay_histogram)[ir]=
+    	     cum_dist_area_with_dist[j-1]+
+    		(cum_dist_area_with_dist[j]-cum_dist_area_with_dist[j-1])*
+    			      (time-tch[j-1])/(tch[j]-tch[j-1]);
+    	break;  /* exits this for loop */
+    	}
+          }
+        }
+      }
+
+    a1=(*time_delay_histogram)[1];
+    sumar=(*time_delay_histogram)[1];
+    (*time_delay_histogram)[1]*=area;
+    
+    if((*num_time_delay_histo_ords)>1)
+      {
+      for(ir=2;ir<=(*num_time_delay_histo_ords);ir++)
+        {
+        a2=(*time_delay_histogram)[ir];
+        (*time_delay_histogram)[ir]=a2-a1;
+        a1=a2;
+        sumar+=(*time_delay_histogram)[ir];
+        (*time_delay_histogram)[ir]*=area;
+        }
+      }
+
+    return;
+} 
+
+// FUNCTION TO INTIALIZE UNSAT STORAGE AND DEFICIT 
+// also intitializes unsaturated zone storage and deficit
+// updates values associated with model structure
+// outputs used in TOPMOD() include sbar, bal, deficit_root_zone
+
+extern void init_water_balance(double **deficit_local, int max_atb_increments, 
```

Comment:
another good example of ordering params and documenting behavior.  Most "init" functions are safe to assume are doing some allocation/initialization, but being clear about WHICH variables are being allocated is important